src/EditorFeatures/Core/ValueTracking/ValueTrackingService.Visitor.cs

```diff
@@ -0,0 +1,266 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ValueTracking
+{
+    internal partial class ValueTrackingService
+    {
+        private class OperationCollector
```

Comment:
was this an effective mechnical transformation?  or where tehre logic changes included?

Reply:
https://github.com/dotnet/roslyn/pull/52361/commits/f66f4c1f60e9401cb2a60fdf97f8c883e3fed5cc was the mechanical transformation, everything after is possibly a logic change.